packages/flutter/lib/src/widgets/animated_raster.dart

```diff
@@ -0,0 +1,215 @@
+class RenderAnimatedRaster extends RenderProxyBox {
+  /// Create a new [RenderAnimatedRaster].
+  RenderAnimatedRaster(this._animation, this._delegate, this._devicePixelRatio) {
+    animation.addListener(markNeedsPaint);
+    animation.addStatusListener(_updateStatus);
+    _updateStatus(animation.status);
+  }
+
+  AnimationStatus _status = AnimationStatus.completed;
+
+  void _updateStatus(AnimationStatus newStatus, [bool painting = false]) {
+    if (newStatus == _status) {
+      return;
+    }
+    _childImage?.dispose();
+    _childImage = null;
+    _status = newStatus;
+    assert(_status == animation.status);
+    if (!painting) {
+      markNeedsPaint();
+    }
+  }
+
+  /// The delegate that controls how the child raster is painted.
+  AnimatedRasterDelegate get delegate => _delegate;
+  AnimatedRasterDelegate _delegate;
+  set delegate(AnimatedRasterDelegate value) {
+    if (value == delegate) {
+      return;
+    }
+    _delegate = value;
+    markNeedsPaint();
+  }
+
+  /// The device pixel ratio used to create the child raster.
+  double get devicePixelRatio => _devicePixelRatio;
+  double _devicePixelRatio;
+  set devicePixelRatio(double value) {
+    if (value == devicePixelRatio) {
+      return;
+    }
+    _devicePixelRatio = value;
+    markNeedsPaint();
+  }
+
+
+  /// The animation used to drive this render object.
+  Animation<double> get animation => _animation;
+  Animation<double> _animation;
+  set animation(Animation<double> value) {
+    if (value == animation) {
+      return;
+    }
+    animation.removeListener(markNeedsPaint);
+    animation.removeStatusListener(_updateStatus);
+    _animation = value;
+    animation.addStatusListener(_updateStatus);
+    animation.addListener(markNeedsPaint);
+    _updateStatus(animation.status);
+    markNeedsPaint();
+  }
+
+  @override
+  void dispose() {
+    animation.removeListener(markNeedsPaint);
+    animation.removeStatusListener(_updateStatus);
+    _childImage?.dispose();
+    _childImage = null;
+    super.dispose();
+  }
+
+  @override
+  void attach(covariant PipelineOwner owner) {
+    animation.addListener(markNeedsPaint);
+    animation.addStatusListener(_updateStatus);
+    _updateStatus(animation.status);
+    super.attach(owner);
+  }
+
+  @override
+  void detach() {
+    animation.removeListener(markNeedsPaint);
+    animation.removeStatusListener(_updateStatus);
+    super.detach();
+  }
+
+  @override
+  bool get isRepaintBoundary => child != null;
+
+  ui.Image? _childImage;
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child == null) {
+      return;
+    }
+    _updateStatus(animation.status, true);
+    switch (_status) {
+      case AnimationStatus.dismissed:
+      case AnimationStatus.completed:
+        super.paint(context, offset);
+        return;
+      case AnimationStatus.forward:
+      case AnimationStatus.reverse:
+        break;
+    }
+    final bool updateImage = delegate.willPaint(animation);
```

Comment:
uber-nit: `updateImage` is a little misleading since it controls whether paint is called?